Lib/python/pyinit.swg

```diff
@@ -200,12 +203,29 @@ swig_varlink_type(void) {
       return NULL;
   }
   return &varlink_type;
+#else
+  PyType_Slot slots[] = {
+    { Py_tp_dealloc, (void*)swig_varlink_dealloc },
+    { Py_tp_getattr, (void*)swig_varlink_getattr },
+    { Py_tp_setattr, (void*)swig_varlink_setattr },
+    { Py_tp_repr, (void*)swig_varlink_repr },
+    { Py_tp_str, (void*)swig_varlink_str },
+    { Py_tp_doc, (void*)varlink__doc__ },
+    { 0, NULL }
+  };
+  PyType_Spec spec = {};
```

Comment:
These empty initializer lists pose a problem for some compilers, notably more recent versions of MSVC, or when using strict compile options (e.g. gcc with `-Wpedantic`).

Reply:
This is indeed not ISO C and we should avoid it:

```
$ echo 'struct { int x; } a = {};'|gcc -pedantic -x c - -o /dev/null
<stdin>:1:23: warning: ISO C forbids empty initializer braces [-Wpedantic]
```

The reason for this is presumably to 0-initialise members not explicitly initialised below.  The [Python docs for this structure](https://docs.python.org/3/c-api/type.html#c.PyType_Spec) explicitly enumerate the members, and the only one not explicitly initialised here is `itemsize`, so we should be able to drop the `{}` and add `spec.itemsize = 0;` below.

There seem to be two more instances of this.